### ISSUE TITLE
Merge operation does not reduce data size

### ISSUE THREAD

**Issue by yashschandra** [xujiajun/nutsdb#77]
**Describe the bug**
Merge operation does not reduce data size, it just copies all datafiles with new IDs.

**To Reproduce**
Steps to reproduce the behavior(Be specific!):
1.   Put same key 1000 times such that it produces multiple datafiles
2.  Check data directory size
3.  Do merge operation
4.  Check data directory size again

Result of step 2 and step 4 are same

Give sample code if you can.

```
func run() {
	opt := nutsdb.DefaultOptions
	opt.Dir = "/tmp/nuts"
	opt.SegmentSize = 1*1024
	db, err := nutsdb.Open(opt)
	if err != nil {
		handleError(err)
	}
	defer db.Close()
	for i:=0; i<1000; i++ {
		update(db, "b1", []byte("hello"), []byte("world"))
	}
	size, err := DirSize("/tmp/nuts")
	if err != nil {
		handleError(err)
	}
	fmt.Println("size ->", size)
	db.Merge()
	size, err = DirSize("/tmp/nuts")
	if err != nil {
		handleError(err)
	}
	fmt.Println("size ->", size)
}

func DirSize(path string) (int64, error) {
	var size int64
	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
		if err != nil {
			return err
		}
		if !info.IsDir() {
			size += info.Size()
		}
		return err
	})
	return size, err
}

func main() {
	run()
}

```

**Expected behavior**
Merge operation should reduce directory size to size of 1 key (since only 1 unique key is present)

**What actually happens**
All datafiles are getting copied as it is with new IDs

**Screenshots**
Above code gives these results -
![image](https://user-images.githubusercontent.com/15684222/100769408-04438b80-3422-11eb-847f-c1ba09440f12.png)

**please complete the following information :**
 - OS: macOS 10.14.6 (18G6042)
      Kernel Version: Darwin 18.7.0
 - NutsDB Version: latest

**Additional context**
Do you want me to raise a PR to fix this?

**Comment by xujiajun** [comment]
@yashschandra  Welcome to submit a PR to fix this.

**Comment by yashschandra** [comment]
@xujiajun check out #83 ?

**Comment by xujiajun** [comment]
@yashschandra Thanks your PR, I have merged your pull request [#83](https://github.com/xujiajun/nutsdb/pull/83) .
